### DISCUSSION TITLE
How can I caculate cache size

### DISCUSSION THREAD

**Discussion by KngStr** [ImageMagick/ImageMagick#3789] [Help] [answered]
Hi,

I'm trying to convert a lot of big tif file.
But some of them take a long time to tell me 'can not extend cache'.
How can I know the cache size before coverting?

Thanks.

**Comment by snibgo** [comment]
IM needs a certain number of bytes per pixel. The number depends on your version of IM, the Q-number and whether you have HDRI.

So, please show the complete text output from "magick -version" or "convert -version".

**Reply by KngStr** [reply]
Thanks. This is my version info 
```
Version: ImageMagick 7.0.11-8 Q16 x64 2021-04-17 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192829913
Features: Cipher DPC OpenCL
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms
 lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zip zlib
```

I downloaded `ImageMagick-7.0.11-8-portable-Q16-x64` form the official site.
OS is Windows 7 x64.

**Comment by snibgo** [comment, marked as answer]
Okay, you have IM v7 Q16, without HDRI. So pixels values are stored as 16-bit integers. V7 means the number of channels is relevant: grayscale is 1 channel, RGB is 3 channels, CMYK is 4 channels, and alpha is an extra channel.

So the pixel cache of an image needs W\*H\*C\*2 bytes, where W=width, H=height, C=channels.

**Reply by KngStr** [reply]
Thanks very much.
I noticed there are two cache files when converting.
Are these files be part of the cache? Or just two cache of original file and converted file?

**Comment by snibgo** [comment]
If your computer has enough memory, the pixel cache is stored there. Otherwise, it is stored on disk, in temporary file(s). Disks are much slower than memory.

I don't know the technical details of disk caches. Perhaps the pixels for the input image are stored in one file, and your conversion involves the temporary creation of another image.

**Reply by KngStr** [reply]
OK. Thanks.
